### PR TITLE
bpo-40528: Improve & clear a bunch of internal asdl thing

### PR DESCRIPTION
**Pull request by isidentical**
- Better error messages with punctuation
- Py_UNREACHABLE() for unreachable states
- Removal of several 'unused' and 'undocumented (in Zephyr ASDL spec) built-in types.

https://bugs.python.org/issue40528

### CODE REVIEW

File: Parser/asdl.py [hunk 1]
```diff
@@ -33,8 +33,7 @@
 # See the EBNF at the top of the file to understand the logical connection
 # between the various node types.
 
-builtin_types = {'identifier', 'string', 'bytes', 'int', 'object', 'singleton',
-                 'constant'}
+builtin_types = {'identifier', 'string', 'int', 'constant'}
```
Comment (serhiy-storchaka) [inline Parser/asdl.py:36]: Update also a comment in Parser/Python.asdl.

I am wondering, should we keep `constant` or `object`?

File: Parser/asdl_c.py [hunk 2]
```diff
@@ -1148,10 +1146,7 @@ def simpleSum(self, sum, name):
             self.emit("Py_INCREF(astmodulestate_global->%s_singleton);" % t.name, 3)
             self.emit("return astmodulestate_global->%s_singleton;" % t.name, 3)
         self.emit("default:", 2)
-        self.emit('/* should never happen, but just in case ... */', 3)
-        code = "PyErr_Format(PyExc_SystemError, \"unknown %s found\");" % name
-        self.emit(code, 3, reflow=False)
-        self.emit("return NULL;", 3)
+        self.emit("Py_UNREACHABLE();", 3);
```
Comment (serhiy-storchaka) [inline Parser/asdl_c.py]: I would remove the `default` cause and add `Py_UNREACHABLE()` after the `switch` statement. The advantage is that the compiler may warn if `switch` does not have `case`s for all enum values and there is no `default`.

**Review by serhiy-storchaka — COMMENTED** [review]
Good work! I was going to remove `bytes` tonight, but all other changes LGTM too.

File: Parser/asdl.py (hunk 1 above)
Comment (isidentical) [reply Parser/asdl.py:36]: neither of them are defined in ASDL spec, but I guess `constant` is more explicit about its job then `object`. So I'd go for keeping it as is, and using `constant`

File: Parser/asdl_c.py (hunk 2 above)
Comment (isidentical) [reply Parser/asdl_c.py]: Didn't know about it. Thanks.

File: Parser/asdl.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Parser/asdl.py:36]: In any case we keep `ast2obj_object()` and `obj2ast_object()`.

**Review by serhiy-storchaka — APPROVED** [review]
:+1:

File: Parser/asdl.py (hunk 1 above)
Comment (isidentical) [reply Parser/asdl.py:36]: Yes, because others are a symlink to it. Do you want me to change constant with object, or is it OK this way?

**Review by pablogsal — APPROVED** [review]
LGTM

Good work @isidentical!

**Pull request merged**